File: account-backend/src/main/java/com/techjava/accountbackend/service/AccountService.java
package com.techjava.accountbackend.service;

import com.techjava.accountbackend.model.Account;

import java.util.List;

public interface AccountService {

    List<Account> getAllAccounts();

    Account getAccountById(int accountId);

    Account addAccount(Account account);

    Account updateAccountById(Account account);

    void deleteAccountById(int accountId);

    List<Account> getAllActiveAccounts();

}

File: src/app/services/account.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Account } from '../model/account';

@Injectable()
export class AccountService {

  baseURL = 'http://localhost:8081/account-portal';

  constructor(private _http: HttpClient) { }

  getAccounts() {
    return this._http.get<Account[]>(this.baseURL + '/all');
  }

  getAccountById(accountId: number) {
    return this._http.get<Account>(this.baseURL + '/id/' + accountId);
  }

  addAccount(account: Account) {
    return this._http.post(this.baseURL + '/add', account);
  }

  updateAccount(account: Account) {
    return this._http.put(this.baseURL + '/update/' + account.accountId, account);
  }

  deleteAccount(accountId: number) {
    return this._http.delete(this.baseURL + '/delete/' + accountId);
  }
}
